modules/study/src/main/BSONHandlers.scala

```diff
@@ -137,85 +142,179 @@ object BSONHandlers:
       comments       = doc.getAsOpt[Comments](F.comments) getOrElse Comments.empty
       gamebook       = doc.getAsOpt[Gamebook](F.gamebook)
       glyphs         = doc.getAsOpt[Glyphs](F.glyphs) getOrElse Glyphs.empty
-      score          = doc.getAsOpt[Score](F.score)
+      eval           = doc.getAsOpt[Score](F.score).map(_.eval)
       clock          = doc.getAsOpt[Centis](F.clock)
-      crazy          = doc.getAsOpt[Crazyhouse.Data](F.crazy)
+      crazyData      = doc.getAsOpt[Crazyhouse.Data](F.crazy)
       forceVariation = ~doc.getAsOpt[Boolean](F.forceVariation)
-    } yield Node(
-      id,
-      ply,
-      Uci.WithSan(uci, san),
-      fen,
-      check,
-      shapes,
-      comments,
-      gamebook,
-      glyphs,
-      score,
-      clock,
-      crazy,
-      Node.emptyChildren,
-      forceVariation
+    yield Branch(
+      id = id,
+      ply = ply,
+      move = Uci.WithSan(uci, san),
+      fen = fen,
+      check = check,
+      shapes = shapes,
+      comments = comments,
+      gamebook = gamebook,
+      glyphs = glyphs,
+      eval = eval,
+      clock = clock,
+      crazyData = crazyData,
+      children = Branches.empty,
+      // TODO we probably need to have access to the variant to set the opening correctly
```

Comment:
I believe we don't save opening to database. So I think we shouldn't change this behavior now, as it will be a breaking change. So, I think it's safe to delete this TODO.

Reply:
It's not about saving the opening in the db but rather lying about the type. `Root` and `Branch` have an `Option[Opening]` field and we're abusing it, since it can mean two things:
- The root/branch has no opening (ie in the middle of a game, or a line that does not exist)
- The root/branch has no opening because we haven't computed it yet. See `toBranch` and `makeRoot` functions in `TreeBuilder`.

We file the opening for now before sending it to the client with those functions but it is not type safe. Ideally we'd remove the opening field and send a `BranchWithOpening` object to the client or something like that but it's not really practical, so I think the best option would be to add right when it's read from the db (without storing it), but by using `Variant.list.openingSensibleVariants(variant) ?? OpeningDb.findByEpdFen(node.fen)` like we do in `TreeBuilder`.

I'm a bit in a hurry and will check later tonight but want to highlight this as I saw thibs started looking at the PR.

Reply:
ah, got it. So, I think, ideally, we should either store it in the database, or remove it from Root/Branch (only compute it when needed).

Computing it right after reading the database seems unnecessary.

Reply:
I don't think we should store it in the database